pennylane/gradients/gradient_transform.py

```diff
@@ -141,21 +141,25 @@ def __init__(self, transform_fn, expand_fn=gradient_expand, differentiable=True,
         self.hybrid = hybrid
         super().__init__(transform_fn, expand_fn=expand_fn, differentiable=differentiable)
 
-    def qnode_execution_wrapper(self, qnode, targs, tkwargs):
+    def default_qnode_wrapper(self, qnode, targs, tkwargs):
         # Here, we overwrite the QNode execution wrapper in order
         # to take into account that classical processing may be present
         # inside the QNode.
         hybrid = tkwargs.pop("hybrid", self.hybrid)
-        _wrapper = super().qnode_execution_wrapper(qnode, targs, tkwargs)
+        _wrapper = super().default_qnode_wrapper(qnode, targs, tkwargs)
         cjac_fn = qml.transforms.classical_jacobian(qnode)
 
         def jacobian_wrapper(*args, **kwargs):
             qjac = _wrapper(*args, **kwargs)
-            cjac = cjac_fn(*args, **kwargs)
 
             if any(m.return_type is qml.operation.Probability for m in qnode.qtape.measurements):
                 qjac = qml.math.squeeze(qjac)
 
+            if not hybrid:

```

Comment:
So basically here `hybrid` is a completely special case to be treated separately?

Reply:
Oh, I forgot to document this change 🙁 

Essentially, the logic was a bit off previously - the classical Jacobian was being computed _always_, even if the user was specifying that they only wanted the quantum-component of the Jacobian. Moving this line up here corrects that --- now, if a user specifies `hybrid=False`, the computation of `cjac` is avoided.